framework/doc/content/source/meshgenerators/FancyExtruderGenerator.md

```diff
@@ -4,7 +4,33 @@
 
 ## Overview
 
-The `FancyExtruderGenerator` adds more customization options over the [MeshExtruderGenerator](MeshExtruderGenerator.md).
+The `FancyExtruderGenerator` adds more customization options over the [MeshExtruderGenerator](MeshExtruderGenerator.md). This mesh generator is also capable of fixing faulty elements on-the-fly during extrusion.
+
+## Multiple Elevations
+
+`FancyExtruderGenerator` extrudes a lower-dimension mesh (1D or 2D) given by [!param](/Mesh/FancyExtruderGenerator/input) into a higher-dimension mesh (2D or 3D) in a direction defined by [!param](/Mesh/FancyExtruderGenerator/direction). The extruded mesh can have multiple elevations with variable extrusion (axial) lengths provided by [!param](/Mesh/FancyExtruderGenerator/heights). Each elevation can have separate subdomains, extra element extra integers, and boundaries defined. The number of axial elements in the different elevations can be provided through [!param](/Mesh/FancyExtruderGenerator/num_layers). Within each elevation, the axial element dimension can be biased using the corresponding growth factor value in [!param](/Mesh/FancyExtruderGenerator/biases).
+
+## Subdomain ID Remapping
+
+By default, the extruded higher-dimension elements retain the same subdomain ids as their original lower-dimension elements. `FancyExtruderGenerator` provides an option to remap subdomain ids for each elevation through [!param](/Mesh/FancyExtruderGenerator/subdomain_swaps), which is a double indexed array input parameter. Each elemental vector of [!param](/Mesh/FancyExtruderGenerator/subdomain_swaps) contains subdomain remapping information for a particular elevation, where the first elemental vector represents the first extruded elevation. The elemental vector contain pairs of subdomain ids: the first subdomain id is the `input` mesh subdomain id that needs to be remapped, and the second subdomain id the new subdomain id to be assigned.
+
+## Extra Element Integer ID Remapping
+
+Extra element integer ID remapping works in a similar manner as subdomain ID remapping. The extra element integers to be remapped must already exist in the `input` mesh and need to be specified in [!param](/Mesh/FancyExtruderGenerator/elem_integer_names_to_swap). As [MOOSE input file syntax](https://mooseframework.inl.gov/application_usage/input_syntax.html) does not allow triple indexed input parameters, the remapping information of multiple extra element integers will be stacked into a double indexed array input parameter ([!param](/Mesh/FancyExtruderGenerator/elem_integer_names_to_swap)). For each extra element integer, the syntax is similar to [!param](/Mesh/FancyExtruderGenerator/subdomain_swaps). The following input example shows the remapping of two extra element integers.
```

Comment:
Seem to need some work in `Parser.C`, I will take a look to see how difficult it would be.

Reply:
https://github.com/idaholab/moose/pull/6459

Similar to this. I guess it would deserve its own issue/PR you are decide to do it.

Reply:
@GiudGiud I looked into this a little more. It seems `libMesh::print_helper()` also needs to be updated to allow triple indexed vector argument. Guess it will definitely be another PR.

Reply:
it s not uncommon to have a fix in libmesh, then do a submodule update in MOOSE
this would really help user input. Matching double stacks and so on is hard to do and hard to verify

Reply:
Okay, let me try.

Reply:
I made it work on my own branch. Will do a libMesh PR and a separate MOOSE PR and then can close this one.

Reply:
https://github.com/libMesh/libmesh/issues/3380
#21930

Reply:
Done